src/DataObjectAnnotator.php

```diff
@@ -60,16 +75,25 @@ class DataObjectAnnotator
 
     /**
      * DataObjectAnnotator constructor.
+     *
      * @throws NotFoundExceptionInterface
+     * @throws ReflectionException
      */
     public function __construct()
     {
         // Don't instantiate anything if annotations are not enabled.
         if (static::config()->get('enabled') === true && Director::isDev()) {
+            $this->extend('beforeDataObjectAnnotator', $this);
```

Comment:
You don't need to pass `$this` - it can be accessed as `$this->owner` from extension instances